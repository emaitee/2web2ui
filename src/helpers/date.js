import moment from 'moment';

const relativeDateOptions = [
  { value: 'hour', label: 'Last Hour' },
  { value: 'day', label: 'Last 24 Hours' },
  { value: '7days', label: 'Last 7 Days' },
  { value: '30days', label: 'Last 30 Days' },
  { value: '90days', label: 'Last 90 Days' },
  { value: 'custom', label: 'Custom' }
];

export {
  getEndOfDay,
  getStartOfDay,
  getRelativeDates,
  relativeDateOptions
};

function getEndOfDay(date) {
  const end = new Date(date);
  end.setHours(23);
  end.setMinutes(59);
  end.setSeconds(59);
  end.setMilliseconds(0);

  return end;
}

function getStartOfDay(date) {
  const start = new Date(date);
  start.setHours(0);
  start.setMinutes(0);
  start.setSeconds(0);
  start.setMilliseconds(0);

  return start;
}

function getRelativeDates(range) {
  const now = moment.utc();
  const to = now.toDate();

  switch (range) {
    case 'hour':
      return { to, from: moment(to).subtract(1, 'hour').toDate() };

    case 'day':
      return { to, from: moment(to).subtract(1, 'day').toDate() };

    case '7days':
      return { to, from: moment(to).subtract(7, 'day').toDate() };

    case '30days':
      return { to, from: moment(to).subtract(30, 'day').toDate() };

    case '90days':
      return { to, from: moment(to).subtract(90, 'day').toDate() };

    default:
      return {};
  }
}
