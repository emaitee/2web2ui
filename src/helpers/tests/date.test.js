import * as dateHelpers from '../date';
import cases from 'jest-in-case';

describe('Date helpers', () => {
  it('should get end of day', () => {
    expect(dateHelpers.getEndOfDay('2017-12-18T10:00')).toEqual(new Date('2017-12-19T04:59:59.000Z'));

  });

  it('should get start of day', () => {
    expect(dateHelpers.getStartOfDay('2017-12-18T10:00')).toEqual(new Date('2017-12-18T05:00:00.000Z'));

  });

  const date = new Date('2017-12-18');
  Date.now = jest.fn(() => date);
  cases('getRelativeDates calculations', ({ range }) => {
    expect(dateHelpers.getRelativeDates(range)).toMatchSnapshot();
  }, {
    'for an hour ago': { range: 'hour' },
    'for a day ago': { range: 'day' },
    'for a week ago': { range: '7days' },
    'for a month': { range: '30days' },
    'for a quarter ago': { range: '90days' },
    'for an invalid range': { range: 'invalid-range' }
  });
});


