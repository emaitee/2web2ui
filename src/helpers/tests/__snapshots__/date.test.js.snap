// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date helpers getRelativeDates calculations for a day ago 1`] = `
Object {
  "from": 2017-12-17T00:00:00.000Z,
  "to": 2017-12-18T00:00:00.000Z,
}
`;

exports[`Date helpers getRelativeDates calculations for a month 1`] = `
Object {
  "from": 2017-11-18T00:00:00.000Z,
  "to": 2017-12-18T00:00:00.000Z,
}
`;

exports[`Date helpers getRelativeDates calculations for a quarter ago 1`] = `
Object {
  "from": 2017-09-18T23:00:00.000Z,
  "to": 2017-12-18T00:00:00.000Z,
}
`;

exports[`Date helpers getRelativeDates calculations for a week ago 1`] = `
Object {
  "from": 2017-12-11T00:00:00.000Z,
  "to": 2017-12-18T00:00:00.000Z,
}
`;

exports[`Date helpers getRelativeDates calculations for an hour ago 1`] = `
Object {
  "from": 2017-12-17T23:00:00.000Z,
  "to": 2017-12-18T00:00:00.000Z,
}
`;

exports[`Date helpers getRelativeDates calculations for an invalid range 1`] = `Object {}`;
